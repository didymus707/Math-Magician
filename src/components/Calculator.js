import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import '../App.css';
import '../css/style.css';
import Nav from './Nav';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = btn => {
    const data = { total, next, operation };
    const results = calculate(data, btn);
    setTotal(results.total);
    setNext(results.next);
    setOperation(results.operation);
  };

  const operations = operation ? next : total;

  return (
    <>
      <Nav />
      <div className="calc-con">
        <div className="math">
          <p>Let&apos;s do some Math!</p>
        </div>
        <div className="cal">
          <Display result={operations || total || next || '0'} />
          <ButtonPanel onBtnClick={handleClick} />
        </div>
      </div>
    </>
  );
};

export default Calculator;
