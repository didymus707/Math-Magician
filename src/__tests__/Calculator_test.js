import React from 'react';
import renderer from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';
import { BrowserRouter, Route } from 'react-router-dom';
import Calculator from '../components/Calculator';

describe('Home', () => {
  it('wraps content within a div with class container', () => {
    const calculator = TestUtils.renderIntoDocument(
      <BrowserRouter>
        <Route exact path="/" component={Calculator} />
      </BrowserRouter>,
    );
    TestUtils.findRenderedDOMComponentWithClass(
      calculator,
      'calc-con',
    );
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Route exact path="/" component={Calculator} />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
