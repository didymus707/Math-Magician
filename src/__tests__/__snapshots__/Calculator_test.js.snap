// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <nav
    className="nav-con"
  >
    <h1>
      Math Magician
    </h1>
    <ul>
      <li
        className="home"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="calc"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calulator
        </a>
      </li>
      <li
        className="quo"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="calc-con"
  >
    <div
      className="math"
    >
      <p>
        Let's do some Math!
      </p>
    </div>
    <div
      className="cal"
    >
      <div
        className="display"
      >
        0
      </div>
      <div
        className="groups"
      >
        <div
          className="row group1"
        >
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="btn btn-color"
            onClick={[Function]}
            type="button"
          >
            /
          </button>
        </div>
        <div
          className="row group2"
        >
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="btn btn-color"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="row group3"
        >
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="btn btn-color"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="row group4"
        >
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="btn btn-color"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="row group5"
        >
          <button
            className="btn btn-wide"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="btn undefined"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="btn btn-color"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
